main.go

```diff
@@ -0,0 +1,268 @@
+		cloudImages.Packet = &packetImage
+	}
+
+	if releaseArch.Media.Openstack != nil {
+		openstack := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Openstack.Artifacts,
+		}
+		artifacts.Openstack = &openstack
+	}
+
+	if releaseArch.Media.Qemu != nil {
+		qemu := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Qemu.Artifacts,
+		}
+		artifacts.Qemu = &qemu
+	}
+
+	if releaseArch.Media.Virtualbox != nil {
+		virtualbox := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Virtualbox.Artifacts,
+		}
+		artifacts.Virtualbox = &virtualbox
+	}
+
+	if releaseArch.Media.Vmware != nil {
+		vmware := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Vmware.Artifacts,
+		}
+		artifacts.Vmware = &vmware
+	}
+
+	if releaseArch.Media.Metal != nil {
+		metal := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Metal.Artifacts,
+		}
+		artifacts.Metal = &metal
+	}
+
+	streamArch := StreamArch{
+		Artifacts: artifacts,
+		Updates:   StreamUpdates{Release: release.Release},
+		Images:    &cloudImages,
+	}
+	return streamArch
+}
+
+func overrideData(original, override interface{}) interface{} {
+	switch override1 := override.(type) {
+	case map[string]interface{}:
+		original1, ok := original.(map[string]interface{})
+		if !ok {
+			return override1
+		}
+		for key, value1 := range original1 {
+			if value2, ok := override1[key]; ok {
+				override1[key] = overrideData(value1, value2)
+			} else {
+				override1[key] = value1
+			}
+		}
+	case nil:
+		original1, ok := original.(map[string]interface{})
+		if ok {
+			return original1
+		}
+	}
+	return override
+}
+
+func run() error {
+	var releasesURL string
+
+	flag.StringVar(&releasesURL, "release-index", "", "Release index location for the required stream")
+	var overrideFilename string
+	flag.StringVar(&overrideFilename, "override", "", "Override file location for the required stream")
+
+	flag.Parse()
+
+	if releasesURL == "" {
+		return errReleaseIndexMissing
+	}
+
+	releasePath, err := ReleaseURL(releasesURL)
+
+	if err != nil {
+		return fmt.Errorf("Error with Release Index: %v", err)
+	}
+
+	var release Release
+	resp, err := http.Get(releasePath)
+	if err != nil {
+		return fmt.Errorf("Error while fetching: %v", err)
+	}
+
+	defer resp.Body.Close()
+
+	decoder := json.NewDecoder(resp.Body)
+	if err = decoder.Decode(&release); err != nil {
+		return fmt.Errorf("Error while decoding json: %v", err)
+	}
+
+	streamArch := make(map[string]*StreamArch)
+	for arch, releaseArch := range release.Architectures {
+		archContent := releaseToStream(releaseArch, release)
+		streamArch[arch] = &archContent
+	}
+
+	streamMetadata := StreamMetadata{
+		Stream:        release.Stream,
+		Metadata:      Metadata{LastModified: time.Now().UTC().Format(time.RFC3339)},
+		Architectures: streamArch,
+	}
+
+	streamFile, err := os.Create("stream.json")
```

Comment:
Minor: should probably make this a switch, and otherwise default to output to the console or something.

Reply:
yeah, sounds good idea. Printing default to stdout now  https://github.com/sinnykumari/fedora-coreos-stream-generator/pull/1/commits/1da0423afb9e2c16bf9b762e71ee5f9ab244aae3